cypress/components/decimal/decimal.cy.tsx

```diff
@@ -268,6 +268,72 @@ context("Tests for Decimal component", () => {
     });
   });
 
+  describe("allowEmptyValue", () => {
+    it.each([
+      [0, "en", "0"],
+      [1, "en", "0.0"],
+      [2, "en", "0.00"],
+      [0, "es-ES", "0"],
+      [1, "es-ES", "0,0"],
+      [2, "es-ES", "0,00"],
+      [0, "fr", "0"],
+      [1, "fr", "0,0"],
+      [2, "fr", "0,00"],
+      [0, "pt-PT", "0"],
+      [1, "pt-PT", "0,0"],
+      [2, "pt-PT", "0,00"],
+    ] as [DecimalProps["precision"], string, string][])(
+      "should format an empty value correctly when precision is %s, locale is %s and allowEmptyValue is false",
+      (precisionValue, localeValue, expectedValue) => {
+        CypressMountWithProviders(
+          <Decimal
+            precision={precisionValue}
+            locale={localeValue}
+            allowEmptyValue={false}
+          />
+        );
+
+        commonDataElementInputPreview().type("100");
+        commonDataElementInputPreview().clear().blur({ force: true });
+        expect(
+          commonDataElementInputPreview().should("have.value", expectedValue)
+        );
+      }
+    );
+
+    it.each([
+      [0, "en", ""],
+      [1, "en", ""],
+      [2, "en", ""],
+      [0, "es-ES", ""],
+      [1, "es-ES", ""],
+      [2, "es-ES", ""],
+      [0, "fr", ""],
+      [1, "fr", ""],
+      [2, "fr", ""],
+      [0, "pt-PT", ""],
+      [1, "pt-PT", ""],
+      [2, "pt-PT", ""],
+    ] as [DecimalProps["precision"], string, string][])(
+      "should format an empty value correctly when precision is %s, locale is %s and allowEmptyValue is true",
```

Comment:
we don't need here a third parameter `""` because we can simply pass the same down below for the assertion and avoid copy/paste in all examples

Reply:
Good point. I copied the format of the set above. Two parameters is sufficient for this and just assert the value is `""` in the actual assertion.